src/metabase/query_analysis/native_query_analyzer.clj

```diff
@@ -289,13 +289,15 @@
   "Returns a set of table identifiers that (may) be referenced in the given card's query.
   Errs on the side of optimism: i.e., it may return tables that are *not* in the query, and is unlikely to fail
   to return tables that are in the query."
-  [driver query & {:keys [mode] :or {mode :ast-walker-1}}]
+  [driver query & {:keys [mode] :or {mode :basic-select}}]
   (let [db-id        (:database query)
         macaw-opts   (nqa.impl/macaw-options driver)
         table-opts   (assoc macaw-opts :mode mode)
         sql-string   (:query (nqa.sub/replace-tags query))
-        parsed-query (macaw/query->tables sql-string table-opts)]
-    (table-refs-for-query parsed-query db-id)))
+        tables       (macaw/query->tables sql-string table-opts)]
+    (if (keyword? tables)
```

Comment:
This "keyword for exceptional cases" is a bit of a pain to propagate without sum types.

Wdyt about going back to exceptions? @tsmacdonald

Reply:
I think it's a @noahmoss / AdminWebapp question about whether they want to consume keywords or exceptions.

If sticking with kws, I would rename `tables` to `tables-or-error` or something

Reply:
Maybe the Malli equivalent of a union type would be helpful here?

Reply:
Either is fine from my side. If exceptions make your life easier I can change the QP code to handle that

Reply:
Went with a tagged-union-esque approach (always a map)